src/gretel_client/projects.py

```diff
@@ -132,34 +157,69 @@ def send(self, data: Union[List[dict], dict]) -> Tuple[list, list]:
 
         Args:
             data: a dict or a list of dicts
+            detection_mode: Determines how to route the record through Gretel's entity
+                detection pipeline. Valid options include "fast", "all" and "none".
+                Selecting "all" will pass the records through a NLP pipeline, but
+                may increase processing latency. Selecting "none" will skip the
+                detection pipeline entirely.
+            headers: (dict) Define additional http request headers.
+            params: (dict) Pass custom request query parameters.
 
         Returns:
             A tuple of (success, failure) lists
         """
-        ret = self.client._write_record_sync(self.name, data)["data"]
+        kwargs = self._patch_params(kwargs, "detection_mode", detection_mode)
+        ret = self.client._write_record_sync(self.name, data, **kwargs)["data"]
         return ret["success"], ret["failure"]
 
-    def send_bulk(self, data: Union[List[dict], dict]) -> "WriteSummary":
+    def _patch_params(self, kwargs: dict, key: str, value: str) -> dict:
+        """Patch in named arguments into query param **kwarg type dictionary"""
+        assert isinstance(kwargs, dict)
```

Comment:
Don't use an assertion outside of a test here, we should check for this not being a `dict` then raise a `ValueError`

Reply:
good catch. fixed.